src/content/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller.md

```diff
@@ -159,11 +159,29 @@ However we could modify the input request or the output given by decorated contr
     }
 ```
 
+In order for the DemoController to work, we also need to implement all the other Action methods we don't interfere with of the decorated controller and simply return the decorated controller method. Example:
+
+```php
+<?php
+ public function anAction($id, Request $request)
+        {
+            return $this->decoratedController->anAction($id, $request);
+        }
+
+```
+
+On some edge cases we might need to perform an Ajax call from our Javascript file (or the Twig template) to our decorating DemoController.
+Curently this will not work though, what we need to do in such a case is to create a regular admincontroller e.g. mymodule/src/Controller/Admin/DemoAjaxController.php with a method, create a route for that method in our mymodule/src/config/routes.yml, generate the URL as described here, [symfony docs](https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/route-generation/) and perform our ajax call using the resulting Symfony route.
```

Comment:
Hello, this sounds like a bug 😄  actually. Can you tell me more? Would you have a code sample I could analyze?

Reply:
Hi,
I'll attach the files here as the repository is private.
Ajax does not work in the decorating RmvProductController but works on AdminActionsController.
Here is the slack thread I have opened for this, [slack](https://prestashop.slack.com/archives/C010UHP5PAR/p1621362812000900l)

[code.zip](https://github.com/PrestaShop/docs/files/6538508/code.zip)

Reply:
If you think seeing the full code will help I could grant you access to my private repo, no problem.

Reply:
Sorry for late answer, week has been very busy I did not have time to explore the topic

Reply:
I understand, no worries, I'll wait!